### PR TITLE
 feat(households): disable households if dwelling no response

### PR DESCRIPTION
**Pull request by maximilianoforlenza**
feat(households): disable households if dwelling no response

Depende de: https://github.com/indec-it/react-native-survey-commons/pull/50

### CODE REVIEW

File: src/components/HouseholdsList/index.js [hunk 1]
```diff
@@ -12,6 +13,8 @@ import matchParamsIdPropTypes from '../../util/matchParamsIdPropTypes';
 import styles from '../AreasList/styles';
 import AddressCard from '../AddressCard';
 
+const householdsEnable = households => reject(households, household => household.disabled);
```
Comment (navarroaxel) [inline src/components/HouseholdsList/index.js]: `getEnabledHouseholds`

File: src/services/survey.js [hunk 2]
```diff
@@ -89,13 +95,28 @@ export default class SurveysService {
         return find(survey.dwellings, dwelling => dwelling.order === dwellingOrder);
     }
 
+    static disableHouseholds(households) {
+        const disable = household => {
+            const newHousehold = household;
+            newHousehold.disabled = true;
+        };
```
Comment (navarroaxel) [inline src/services/survey.js]: llevar afuera de la clase del servicio y arriba.

File: src/services/survey.js [hunk 3]
```diff
@@ -78,8 +78,14 @@ export default class SurveysService {
...
-        const maxOrder = max(map(dwelling.households, household => household.order)) || 0;
-        dwelling.households.push(new Household({order: maxOrder + 1}));
+        const maxOrder = max(
+            map(
+                reject(
+                    dwelling.households, household => household.disabled
+                ), household => household.order
+            )
+        ) || 0;
+        dwelling.households.push(new Household({order: maxOrder + 1, id: Date.now()}));
```
Comment (navarroaxel) [inline src/services/survey.js]: el `id` es inicializado por la clase `Household`

**Comment by maximilianoforlenza** [comment]
listo

File: src/components/HouseholdsList/index.js [hunk 4]
```diff
@@ -98,7 +98,7 @@ class HouseholdsList extends Component {
                 />
                 <Title>Listado de hogares</Title>
                 <View style={styles.tableContainer}>
-                    <Table columns={this.columns} data={dwelling.households}/>
+                    <Table columns={this.columns} data={new Dwelling(dwelling.households).getEnabledHouseholds()}/>
```
Comment (navarroaxel) [inline src/components/HouseholdsList/index.js]: `dwelling.getHouseholds()`

File: src/services/survey.js [hunk 5]
```diff
@@ -78,7 +83,13 @@ export default class SurveysService {
      * @returns {Dwelling} The given dwelling including the new household.
      */
     static addHouseholdToDwelling(dwelling) {
-        const maxOrder = max(map(dwelling.households, household => household.order)) || 0;
+        const maxOrder = max(
+            map(
+                reject(
+                    dwelling.households, household => household.disabled
```
Comment (navarroaxel) [inline src/services/survey.js]: `dwelling.getHouseholds()`

File: src/services/survey.js [hunk 6]
```diff
@@ -89,6 +100,14 @@ export default class SurveysService {
         return find(survey.dwellings, dwelling => dwelling.order === dwellingOrder);
     }
 
+    static disableHouseholds(households) {
+        if (isArray(households)) {
+            forEach(households, household => disableHousehold(household));
```
Comment (navarroaxel) [inline src/services/survey.js]: `forEach(households, disableHousehold);`

File: src/services/survey.js [hunk 7]
```diff
@@ -89,6 +100,14 @@ export default class SurveysService {
         return find(survey.dwellings, dwelling => dwelling.order === dwellingOrder);
     }
 
+    static disableHouseholds(households) {
+        if (isArray(households)) {
```
Comment (navarroaxel) [inline src/services/survey.js]: `forEach(castArray(households), disableHousehold);`

File: src/services/survey.js [hunk 8]
```diff
@@ -106,7 +128,7 @@ export default class SurveysService {
         const dwellingOrder = toNumber(dwelling);
         const survey = await SurveysService.findById(id);
         const foundDwelling = find(survey.dwellings, d => d.order === dwellingOrder);
-        return foundDwelling.households;
+        return new Dwelling(foundDwelling.households).getEnabledHouseholds();
```
Comment (navarroaxel) [inline src/services/survey.js]: `new Dwelling(foundDwelling).getHouseholds()`

**Comment by maximilianoforlenza** [comment]
listo

**Pull request merged**